### ISSUE TITLE
feature: add support for Dokka in the JavaDoc integration

### ISSUE THREAD

**Issue by pauliesnug** [dzikoysk/reposilite#2117]
### Request details

Currently, if a `.jar` file ends with `-javadoc.jar`, it can be viewed through the Reposilite web UI. Trying to do this with a `-dokka.jar`, [Kotlin's documentation engine](https://github.com/Kotlin/dokka) doesn't seem to work. Is it possible for the JavaDoc detection to include files ending with `-dokka.jar`? Or would more support need to be added to support Dokka due to a different file structure?

**Comment by dzikoysk** [comment]
I honestly don't know, as I never used Dokka 😅 But yeah, it makes sense to support it at some point.

**Comment by pauliesnug** [comment]
> I honestly don't know, as I never used Dokka 😅 But yeah, it makes sense to support it at some point.

I wonder if it would work if we just changed the artifact ID to end with `-javadoc.jar`

**Comment by RedstoneWizard08** [comment]
> > I honestly don't know, as I never used Dokka 😅 But yeah, it makes sense to support it at some point.
> 
> I wonder if it would work if we just changed the artifact ID to end with `-javadoc.jar`

Not quite. Reposilite's access control breaks some stuff and things don't render properly.

**Comment by pauliesnug** [comment]
> > > I honestly don't know, as I never used Dokka 😅 But yeah, it makes sense to support it at some point.
> > 
> > 
> > I wonder if it would work if we just changed the artifact ID to end with `-javadoc.jar`
> 
> Not quite. Reposilite's access control breaks some stuff and things don't render properly.

Oh cool! Thank you so much for the PR! I'd love Dokka support so much, our team could really use it

**Comment by RedstoneWizard08** [comment]
Ok, scratch ALL of that, I found the issue.
![image](https://github.com/dzikoysk/reposilite/assets/94275204/7829dc0f-cc70-482b-ba0d-e0e9c85f4592)

**Comment by TOTHT0MI** [comment]
Dokka by default generates a HTML documentation, so it's only a static-site serving task.
However it can also output in javadoc format and Markdown. The later requires a markdown engine in the frontend, if we wish to support all formats.

See HTML format here (sorry for redaction, don't wanna plug other projects):
![image](https://github.com/user-attachments/assets/f4ffcbce-0040-4632-bc85-04b78616f237)
